toolchain/lower/lower.h

```diff
@@ -0,0 +1,38 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef CARBON_TOOLCHAIN_LOWER_LOWER_H_
+#define CARBON_TOOLCHAIN_LOWER_LOWER_H_
+
+#include "common/ostream.h"
+#include "llvm/IR/LLVMContext.h"
+#include "llvm/IR/Module.h"
+#include "toolchain/semantics/semantics_ir.h"
+
+namespace Carbon {
+
+// The output of lowering.
+struct LowerResult {
+  LowerResult(llvm::LLVMContext& llvm_context, llvm::StringRef module_id)
+      : module(std::make_unique<llvm::Module>(module_id, llvm_context)) {}
+
+  // Prints the module's IR.
+  auto Print(llvm::raw_ostream& output) const -> void;
+
+  // llvm::Module isn't copyable or moveable; unique_ptr allows easier handling.
+  std::unique_ptr<llvm::Module> module;
+
+  bool has_errors = false;
+};
+
+// Lowers Semantics IR to LLVM IR.
+class Lower {
+ public:
+  static auto Make(llvm::LLVMContext& llvm_context,
+                   const SemanticsIR& semantics_ir) -> LowerResult;
+};
```

Comment:
Similar to the above, not really important to address here, but what do you think about this just being a function that returns an `llvm::Module`, and potentially naming it more `LowerToLLVM`?

Somewhat, there are two questions here:

1) Will we want a stateful entity that can emit multiple LLVM modules (from multiple inputs)?
2) Will we want to include any extra data besides the actual IR?

I somewhat strongly suspect the answer to (2) is "no" for now, and it seems reasonably doubtful it will become "yes" in even the medium term.

For (1), I'm less confident, but I think the answer again might be "no". There are things I can imagine caching here, but I'm worried they would be relatively low value and relatively risky for bugs. Most of the high value things that you might imagine caching would already be handled by the `LLVMContext` input.

Anyways, its easy to change all of these, so just recording thoughts for now. =]